src/ZplBuilder.php

```diff
@@ -88,6 +88,27 @@ public function setFont(string $font, float $size, ?float $width = null): void
         $this->commands[] = $command;
     }
 
+    public function setQuantity(int $quantity, int $pauseQty = 0, int $replicate = 1): void
+    {
+        $this->commands[] = '^PQ' . $quantity . ',' . $pauseQty . ',' . ($replicate <= 0 ? 1 : $replicate);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see \Zpl\AbstractBuilder::drawText()
+     */
+    public function drawSerialNumber(float $x, float $y, string $text, string $orientation = 'N', bool $invert = false, int $step = 1, bool $ceroFill = true): void
+    {
+        $this->commands[] = '^FW' . $orientation;
+        $this->commands[] = '^FO' . $this->toDots($x) . ',' . $this->toDots($y);
+        if ($invert === true) {
+            $this->commands[] = '^FR';
+        }
+        $this->commands[] = '^SN' . $text . ',' . ($step <= 0 ? 1 : $step) . ',' . ($ceroFill === true ? 'Y' : 'N') . '^FS';
+        $this->commands[] = '^FWN';
+    }
```

Comment:
Can we control the orientation outside? Returning to normal after the call could lead to unwanted results if we manually set the orientation outside at some point.

```suggestion
    public function drawSerialNumber(float $x, float $y, int $start = 1, int $step = 1, bool $pad = true, bool $invert = false): void
    {
        $this->commands[] = '^FO' . $this->toDots($x) . ',' . $this->toDots($y);
        if ($invert === true) {
            $this->commands[] = '^FR';
        }
        $this->commands[] = '^SN' . $start . ',' . ($step <= 0 ? 1 : $step) . ',' . ($pad ? 'Y' : 'N') . '^FS';
    }
```